### PR TITLE
feat(exporter): Add wrapper for googlemanagedprometheus to allow for additional credential options

### PR DESCRIPTION
**Pull request by BinaryFissionGames**
### Proposed Change
* Create a wrapper, similar to the GCP wrapper, that allows for `credentials` and `credential_file` options.

I've kept this more reduced in scope than the GCP wrapper, only adding in the credentials functionality, as well as the project detection functionality.

##### Checklist
- [x] Changes are tested
- [ ] CI has passed

### CODE REVIEW

File: exporter/googlemanagedprometheusexporter/config.go [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+type Config struct {
+	Credentials     string      `mapstructure:"credentials"`
+	CredentialsFile string      `mapstructure:"credentials_file"`
+	GMPConfig       *gmp.Config `mapstructure:",squash"`
+}
+
+// Validate validates the config
+func (c *Config) Validate() error {
+	var err error
+	err = multierr.Append(err, c.GMPConfig.Validate())
```
Comment (cpheps) [inline exporter/googlemanagedprometheusexporter/config.go:44]: nit: Do we need to use multi err here as there's only one error?

**Review by cpheps — APPROVED** [review]
LGTM

File: exporter/googlemanagedprometheusexporter/config.go (hunk 1 above)
Comment (BinaryFissionGames) [reply exporter/googlemanagedprometheusexporter/config.go]: Nope, totally missed that as I was cutting out some of the logic. Good catch!

**Review by jsirianni — APPROVED** [review]
Working great for me

File: exporter/googlemanagedprometheusexporter/README.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+|--------------------|---------|-----------------------|----------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `credentials_file` | string  |                       | `false`  | The credentials file used to authenticate the GPM client. Ignored if `credentials` is set.                                                                                                                                                  |
+| `credentials`      | string  |                       | `false`  | The credentials JSON used to authenticate the GPM client.                                                                                                                                                                                   |
```
Comment (Mrod1598) [inline exporter/googlemanagedprometheusexporter/README.md:18]: might be worth linking out to google for credentials

**Review by Mrod1598 — APPROVED** [review]

**Pull request merged**